scenes/SceneFilesFolder.js

```diff
@@ -15,9 +21,138 @@ import EmptyState from "~/components/core/EmptyState";
 
 const POLLING_INTERVAL = 10000;
 
+const STYLES_FILTERS_CONTAINER = css`
+  height: 60px;
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-end;
+  align-items: flex-start;
+`;
+
+const STYLES_TOOLTIP_ANCHOR = css`
+  border: 1px solid #f2f2f2;
+  background-color: ${Constants.system.white};
+  border-radius: 4px;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: flex-start;
+  box-shadow: 0px 8px 24px rgba(178, 178, 178, 0.2);
+  position: absolute;
+  z-index: ${Constants.zindex.tooltip};
+`;
+
+const STYLES_FILETYPE_TOOLTIP = css`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-around;
+  height: 260px;
+`;
+
+const STYLES_PRIVACY_TOOLTIP = css`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-around;
+  height: 115px;
+  font-family: ${Constants.font.medium};
+  font-size: 16px;
+`;
+
 export default class SceneFilesFolder extends React.Component {
   state = {
     view: 0,
+    filetypeTooltip: false,
+    privacyTooltip: false,
+    filters: {},
+    filtersActive: false,
+    privacy: "All",
+    files: this.props.viewer.library[0].children,
+  };
+
+  _handleFiletypeTooltip = () => {
+    this.setState({ filetypeTooltip: !this.state.filetypeTooltip });
+  };
+
+  _handlePrivacyTooltip = () => {
+    this.setState({ privacyTooltip: !this.state.privacyTooltip });
+  };
+
+  _handlePrivacyFilter = (filter) => {
+    const viewer = this.props.viewer;
+    const filtered = getPublicAndPrivateFiles({ viewer });
+    if (filter === "All") {
+      this.setState({ privacy: "All" }, this._filterFiles);
```

Comment:
Represent these states in all lowercase or uppercase to save yourself the one by one grief you could have in the future.

Reply:
they're sentence-cased for the button so I can just put {this.state.privacy} without manipulating the text, it has to be done ;-;